pkg/reconciler/pipelinerun/affinity_assistant.go

```diff
@@ -56,73 +56,74 @@ func (c *Reconciler) createOrUpdateAffinityAssistants(ctx context.Context, wb []
 	var errs []error
 	var unschedulableNodes sets.Set[string] = nil
 	for _, w := range wb {
-		if w.PersistentVolumeClaim != nil || w.VolumeClaimTemplate != nil {
-			affinityAssistantName := getAffinityAssistantName(w.Name, pr.Name)
-			a, err := c.KubeClientSet.AppsV1().StatefulSets(namespace).Get(ctx, affinityAssistantName, metav1.GetOptions{})
-			claimName := getClaimName(w, *kmeta.NewControllerRef(pr))
-			switch {
-			// check whether the affinity assistant (StatefulSet) exists or not, create one if it does not exist
-			case apierrors.IsNotFound(err):
-				affinityAssistantStatefulSet := affinityAssistantStatefulSet(affinityAssistantName, pr, claimName, c.Images.NopImage, cfg.Defaults.DefaultAAPodTemplate)
-				_, err := c.KubeClientSet.AppsV1().StatefulSets(namespace).Create(ctx, affinityAssistantStatefulSet, metav1.CreateOptions{})
+		if w.PersistentVolumeClaim == nil && w.VolumeClaimTemplate == nil {
+			continue
+		}
+
+		var claimTemplates []corev1.PersistentVolumeClaim
+		var claims []corev1.PersistentVolumeClaimVolumeSource
+		if w.PersistentVolumeClaim != nil {
+			claims = append(claims, *w.PersistentVolumeClaim.DeepCopy())
+		} else if w.VolumeClaimTemplate != nil {
+			claimTemplate := w.VolumeClaimTemplate.DeepCopy()
+			claimTemplate.Name = volumeclaim.GetPersistentVolumeClaimNameWithoutAffinityAssistant(w.VolumeClaimTemplate.Name, w, *kmeta.NewControllerRef(pr))
```

Comment:
I think the separate functions you created definitely help with readability! The one confusing bit is here-- the fact that `GetPersistentVolumeClaimNameWithoutAffinityAssistant` is called in the function `createOrUpdateAffinityAssistants`. Maybe a clarifying comment would help here?

Reply:
Happy to see it helps with readability! Yeah, added a comment here.